### ISSUE TITLE
not compatible with "kirkstone"

### ISSUE THREAD

**Issue by BennyBread** [RDunkley/meta-dotnet-core#37]
Hi I try to use this for an raspberrypi build but it states that the layer it's not compatible with kirstone.

I'm a yocto beginner.

Is it possible to support/create a kirkstone branch?

Is it enough to change layer.conf like this?

`LAYERSERIES_COMPAT_dotnet-core = "`**kirkstone**` rocko sumo thud warrior zeus dunfell"`

**Comment by RDunkley** [comment]
Use the Honister branch and then add Kirkwood to the compatible layer. That
should do it unless there are breaking changes.

**Comment by leoguiders** [comment]
It looks like dotnet-core is not compatible with version 2.13 of lttng (hardknott/lttng-2.12 had a .0 module, 2.13 changed to .1 ABI).

**Comment by RDunkley** [comment]
Is this with version 6.0.8 of the runtime?

**Comment by leoguiders** [comment]
I tried 6.0.9 but I do not expect the minor version to make any difference

As far as I can tell dotnet would have to be recompiled with 2.13 to be compatible but I do not have much experience with library versioning. Also I was unable to use the 2.12 recipes with kirkstone

See https://github.com/dotnet/runtime/issues/57784

**Comment by RDunkley** [comment]
You could try creating a link from the old file version name to the new to
trick dotnet into loading the new library as the old. As long as the
signatures for the functions haven't changed then that should work.

I thought .net dropped the dependency on lttng in version 6, but maybe they
did only the native side, but are still pinvoking into it on the managed
side. I don't have a good way of scanning the .net assemblies for pinvokes
when i do a new version like i can for the native library dependencies.

**Comment by leoguiders** [comment]
The lttng changelog says for v2.13:

> Bumping the soname major to 1, will require applications and tracepoint
> providers to be rebuilt against an updated LTTng-UST to use it.
> 
> Old applications and tracepoint providers linked against libraries with
> major soname 0 should be able to co-exist on the same system.

I guess this means I can either recompile dotnet with the version 1 .so or build a version 0 .so and deploy it in parallel to the v1 so.

The bitbake process does seem to scan for required library dependencies.

**Comment by leoguiders** [comment]
I was finally able to install aspnet-core using the hack implemented here: https://github.com/DynamicDevices/meta-mono/blob/master/recipes-mono/dotnet/dotnet_6.0.302.bb

```
    # Hack to fix liblttng-ust dependency issues
    patchelf --remove-needed liblttng-ust.so.0 ${D}${datadir}/dotnet/shared/Microsoft.NETCore.App/${PV}/libcoreclrtraceptprovider.so
```

**Comment by electroScorpion** [comment]
Pull request was created. [GamerBroGebauer:Honister](https://github.com/RDunkley/meta-dotnet-core/pull/45).

**Comment by RDunkley** [comment]
Merged into 'kirkstone' branch with 4d5c894b2aa7876f91b34d4acf45ed98aaee7ac7
